Content.Server/Explosion/EntitySystems/TriggerSystem.Signal.cs

```diff
@@ -0,0 +1,29 @@
+using Content.Server.Explosion.Components;
+using Content.Server.MachineLinking.Events;
+using Content.Server.MachineLinking.Components;
+
+namespace Content.Server.Explosion.EntitySystems
+{
+    public sealed partial class TriggerSystem
+    {
+        private void InitializeSignal()
+        {
+            SubscribeLocalEvent<SignalTriggerComponent,SignalReceivedEvent>(OnSignalReceived);
+            SubscribeLocalEvent<SignalTriggerComponent,ComponentInit>(OnInit);
+        }
+
+        private void OnSignalReceived(EntityUid uid, SignalTriggerComponent component, SignalReceivedEvent args)
+        {
+            if (args.Port != "Trigger")
+                return;
+
+            Trigger(uid);
+        }
+        private void OnInit(EntityUid uid, SignalTriggerComponent component, ComponentInit args)
+        {
+            var receiver = EnsureComp<SignalReceiverComponent>(uid);
+            if (!receiver.Inputs.ContainsKey("Trigger"))
+                receiver.AddPort("Trigger");
+        }
```

Comment:
EnsureComp should be avoided unless you reaaallllyyy need it. Constants should also be used for the strings.

Also can ports be removed? If so should handle shutdown as well.

Reply:
multiple components might use the same port. So realistically it would need to use remove-port-attempt events, which should be done but is probably out of scope

Reply:
I've checked with other people, since I'm not the brightest or adept in any fucking way at all with this stuff. We tried changing this but it broke the build with out mthod. After other people conversed, Rane, Moony, others, (do not ask me to make sense of it), we concluded to go back to using this. It works. 

We'd like to override and use EnsureComp here.

Reply:
On the ensure comp stuff, I agree that it should be ensure comp in this case. 

For most components, if they rely on some other component then that component should just be included in the relevant entity prototype. But for something like machine linking or triggering, where the components might be dynamically added to an entity due to a player crafting/modifying some device, each component should have init/add methods that ensure that it will be functional.

Reply:
sgtm